packages/kbn-unified-field-list/src/containers/unified_field_list_sidebar/field_list_sidebar.tsx

```diff
@@ -226,24 +205,38 @@ export const UnifiedFieldListSidebarComponent: React.FC<UnifiedFieldListSidebarP
       [searchMode, stateService.creationOptions.disableMultiFieldsGroupingByParent]
     );
 
-  const { fieldListFiltersProps, fieldListGroupedProps } = useGroupedFields<DataViewField>({
-    dataViewId: (searchMode === 'documents' && dataView?.id) || null, // passing `null` for text-based queries
-    allFields,
-    popularFieldsLimit:
-      searchMode !== 'documents' || stateService.creationOptions.disablePopularFields
-        ? 0
-        : popularFieldsLimit,
-    isAffectedByGlobalFilter,
-    services: {
-      dataViews,
-      core,
-    },
-    sortedSelectedFields: onSelectedFieldFilter ? undefined : selectedFieldsState.selectedFields,
-    onSelectedFieldFilter,
-    onSupportedFieldFilter:
-      stateService.creationOptions.onSupportedFieldFilter ?? onSupportedFieldFilter,
-    onOverrideFieldGroupDetails: stateService.creationOptions.onOverrideFieldGroupDetails,
-  });
+  const { fieldListFiltersProps, fieldListGroupedProps, allFieldsModified } =
+    useGroupedFields<DataViewField>({
+      dataViewId: (searchMode === 'documents' && dataView?.id) || null, // passing `null` for text-based queries
+      allFields,
+      popularFieldsLimit:
+        searchMode !== 'documents' || stateService.creationOptions.disablePopularFields
+          ? 0
+          : popularFieldsLimit,
+      isAffectedByGlobalFilter,
+      services: {
+        dataViews,
+        core,
+      },
+      sortedSelectedFields: onSelectedFieldFilter ? undefined : selectedFieldsState.selectedFields,
+      onSelectedFieldFilter,
+      onSupportedFieldFilter:
+        stateService.creationOptions.onSupportedFieldFilter ?? onSupportedFieldFilter,
+      onOverrideFieldGroupDetails: stateService.creationOptions.onOverrideFieldGroupDetails,
+      getNewFieldsBySpec: (fieldSpecArr) =>
```

Comment:
Let's extract this handler function separately so it does not change.

Reply:
Makes sense and done ,thx!

Reply:
Same for Lens

Reply:
also, done, thx!